src/averageAges.js

```diff
@@ -52,7 +64,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrChildren = people.filter(child => (!onlyWithSon
+    ? people.some(mum => mum.name === child.mother)
+    : people.some(mum => mum.name === child.mother) && (child.sex === 'm')
+  ));
+
+  const getDifferentMotherChildAges = arrChildren.map(child => child.born
+    - (people.find(mum => mum.name === child.mother).born));
```

Comment:
```suggestion
  const getDifferentMotherChildAges = arrChildren.map(child => 
    child.born - (people.find(mum => mum.name === child.mother).born));
```

It's better not to translate arithmetical operators to the next line. In cases when the line is too long better to use formatting like this (next line after arrow function sign)